test/suites/smoke/test-polkadot-decoding.ts

```diff
@@ -7,6 +7,64 @@ import { fail } from "assert";
 
 const pageSize = (process.env.PAGE_SIZE && parseInt(process.env.PAGE_SIZE)) || 500;
 
+const extractStorageKeyComponents = (storageKey: string) => {
+  // The full storage key is composed of
+  // - The 0x prefix (2 characters)
+  // - The module prefix (32 characters)
+  // - The method name (32 characters)
+  // - The parameters (variable length)
+  const moduleKey = storageKey.substring(0, 2 + 32); // Includes the 0x prefix
+  const fnKey = storageKey.substring(2 + 32, 2 + 32 + 32);
+  const paramsKey = storageKey.substring(2 + 32 + 32);
```

Comment:
This can be improved:
```suggestion
const regex = /0x(?<module>[a-f0-9]{32})(?<fn>[a-f0-9]{32})(?<params>[a-f0-9]*)/g;
const match = regex.exec(storageKey);
if (!match) {
  throw new Error("Invalid storage key");
}
const { module, fn, params } = match.groups;
```

Reply:
Current suggestion does not work as is, but will replace the substrings with regex and commit changes. I like adding some sanitization to the keys.

Reply:
It should work, I have tested it

Reply:
Added the regex :+1: